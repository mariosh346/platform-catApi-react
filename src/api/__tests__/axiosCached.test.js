import { describe, it, expect, vi, beforeEach } from 'vitest'
import { setupCache } from 'axios-cache-interceptor'
import axios from 'axios'
vi.mock('axios')

const mockedAxios = vi.mocked(axios, { deep: true })
const mockedAxiosCreate = vi.fn(val => val)
mockedAxios.create = mockedAxiosCreate

vi.mock('axios-cache-interceptor', () => ({
  setupCache: vi.fn(() => ({
    cached: true,
    get: vi.fn(),
  })),
}))

beforeEach(() => {
  vi.clearAllMocks()
})

describe('getAxios', () => {
  it('should create a cached axios instance with axios and the correct ttl', async () => {
    const { getAxios } = await import('../axiosCached.ts')
    const cachedInstance = getAxios()
    expect(cachedInstance).toHaveProperty('cached', true)
    const calls = vi.mocked(setupCache).mock.calls[0]
    expect(calls).toMatchSnapshot()
  })
})
