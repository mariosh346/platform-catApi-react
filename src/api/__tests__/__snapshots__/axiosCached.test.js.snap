// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getAxios > should create a cached axios instance with axios and the correct ttl 1`] = `
[
  {
    "baseURL": "https://api.thecatapi.com/v1",
  },
  {
    "ttl": 3600000,
  },
]
`;
